lib/shoryuken/queue.rb

```diff
@@ -41,23 +38,55 @@ def receive_messages(options)
         map { |m| Message.new(client, self, m) }
     end
 
+    # Returns whether this queue is a FIFO queue or not.
+    # @return [TrueClass, FalseClass]
+    def is_fifo?
+      queue_attributes.attributes[FIFO_ATTRIBUTE] == 'true'
+    end
+
+    # Returns whether this queue has content based deduplication enabled or not.
+    # @return [TrueClass, FalseClass]
+    def has_content_deduplication?
+      queue_attributes.attributes[CONTENT_DEDUP_ATTRIBUTE] == 'true'
+    end
+
     private
 
+    FIFO_ATTRIBUTE = 'FifoQueue'
```

Comment:
@richseviora I'm getting this:

```shell
     Aws::SQS::Errors::InvalidAttributeName:
       Unknown Attribute FifoQueue.
```

while trying to run `SPEC_ALL=true rspec` which performs an integration test end-to-end. I think that's because the idea of the generic `queue_attributes`, it's also requesting `FifoQueue` for non-FIFO queues.

I think we should use:

```ruby
# See https://docs.aws.amazon.com/sdkforruby/api/Aws/SQS/Client.html#get_queue_attributes-instance_method
client.get_queue_attributes(queue_url: url, attribute_names: ['All'])
```

So it would return `FifoQueue` when available. WDYT?